query/src/main/java/org/infinispan/search/mapper/mapping/impl/InfinispanMapping.java

```diff
@@ -95,6 +96,11 @@ public <E> SearchScope<E> scope(Collection<? extends Class<? extends E>> targete
       return createScope(targetedTypes);
    }
 
+   @Override
+   public Optional<SearchScope<?>> findScopeAll() {
+      return Optional.of(doCreateScope(typeContextContainer.allTypeIdentifiers()));
+   }
+
```

Comment:
If I remember correctly, @wburns usually suggests not to use Optional.

Reply:
@wburns is right. Personally, the code looks better with `Optional` than `null` checks, but `Optional` has a higher memory cost since the JVM has to allocate (and garbage collect) the memory for the object.
This method is only used by 2 "events": `cache.clear()` and topology change and I consider both a rare event...

If you or @wburns think otherwise, I don't mind removing the `Optional` and replacing it with `null` checks.

Reply:
I think that we can live with it.